energy-integration/farm-boosted-yields/src/lib.rs

```diff
@@ -44,31 +54,41 @@ pub trait FarmBoostedYieldsModule:
     + weekly_rewards_splitting::update_claim_progress_energy::UpdateClaimProgressEnergyModule
     + energy_query::EnergyQueryModule
 {
+    #[only_owner]
     #[endpoint(collectUndistributedBoostedRewards)]
-    fn collect_undistributed_boosted_rewards(&self) {
-        self.require_caller_has_admin_permissions();
-
-        let collect_rewards_offset = USER_MAX_CLAIM_WEEKS + 1usize;
+    fn collect_undistributed_boosted_rewards(
+        &self,
+        start_week: Week,
+        end_week: Week,
+        dest_address: ManagedAddress,
```

Comment:
We don't need this, we will use the address saved in the energy factory (multisig).